### PR TITLE
Initial Azure.ResourceManager.AppPlatform

### LINKED ISSUE #29037: Initial Azure.ResourceManager.AppPlatform

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/api/Azure.ResourceManager.AppPlatform.netstandard2.0.cs [hunk 1]
```diff
@@ -0,0 +1,2133 @@
...
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.AppPlatform.CustomDomainResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.AppPlatform.CustomDomainResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.AppPlatform.CustomDomainResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.AppPlatform.CustomDomainResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class CustomDomainResourceData : Azure.ResourceManager.Models.ResourceData
+    {
+        public CustomDomainResourceData() { }
+        public Azure.ResourceManager.AppPlatform.Models.CustomDomainProperties Properties { get { throw null; } set { } }
+    }
+    public partial class DeploymentResource : Azure.ResourceManager.ArmResource
```
Comment (fengzhou-msft) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/api/Azure.ResourceManager.AppPlatform.netstandard2.0.cs:536]: `DeploymentResource` is too generic and maybe use `AppDeploymentResource`.

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/api/Azure.ResourceManager.AppPlatform.netstandard2.0.cs [hunk 2]
```diff
@@ -0,0 +1,2133 @@
...
+    }
+    public partial class SupportedStackResourceData : Azure.ResourceManager.Models.ResourceData
+    {
+        public SupportedStackResourceData() { }
+        public Azure.ResourceManager.AppPlatform.Models.SupportedStackResourceProperties Properties { get { throw null; } set { } }
+    }
+}
+namespace Azure.ResourceManager.AppPlatform.Models
+{
+    public partial class ActiveDeploymentCollection
```
Comment (fengzhou-msft) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/api/Azure.ResourceManager.AppPlatform.netstandard2.0.cs:925]: Avoid using `Collection` as model suffix.

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/api/Azure.ResourceManager.AppPlatform.netstandard2.0.cs [hunk 3]
```diff
@@ -0,0 +1,2133 @@
...
+        public string IssuedDate { get { throw null; } }
+        public string Issuer { get { throw null; } }
+        public string SubjectName { get { throw null; } }
+        public string Thumbprint { get { throw null; } }
+    }
+    public partial class CheckNameAvailabilityContent
+    {
+        public CheckNameAvailabilityContent(string resourceType, string name) { }
+        public string Name { get { throw null; } }
+        public string ResourceType { get { throw null; } }
```
Comment (fengzhou-msft) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/api/Azure.ResourceManager.AppPlatform.netstandard2.0.cs:1264]: change type from `string` to `ResorceType?`

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/ResourceSku.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+#nullable disable
+
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.ResourceManager.AppPlatform.Models
+{
+    /// <summary> Describes an available Azure Spring Cloud SKU. </summary>
+    public partial class ResourceSku
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/ResourceSku.cs:14]: `ResourceSku` seems to be too general and might cause confusion with the sku in Resources RP.

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/ResourceSku.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        /// <summary> Gets the type of resource the SKU applies to. </summary>
+        public string ResourceType { get; }
+        /// <summary> Gets the name of SKU. </summary>
+        public string Name { get; }
+        /// <summary> Gets the tier of SKU. </summary>
+        public string Tier { get; }
+        /// <summary> Gets the capacity of SKU. </summary>
+        public SkuCapacity Capacity { get; }
+        /// <summary> Gets the set of locations that the SKU is available. </summary>
+        public IReadOnlyList<string> Locations { get; }
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/ResourceSku.cs:55]: `AzureLocation`?

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/SsoProperties.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            Scope = scope;
+            ClientId = clientId;
+            ClientSecret = clientSecret;
+            IssuerUri = issuerUri;
+        }
+
+        /// <summary> It defines the specific actions applications can be allowed to do on a user&apos;s behalf. </summary>
+        public IList<string> Scope { get; }
+        /// <summary> The public identifier for the application. </summary>
+        public string ClientId { get; set; }
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/SsoProperties.cs:39]: Should be `Guid`?

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/api/Azure.ResourceManager.AppPlatform.netstandard2.0.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/appplatform/Azure.ResourceManager.AppPlatform/api/Azure.ResourceManager.AppPlatform.netstandard2.0.cs:925]: too many for manual fixing, so would like the codegen can auto-resolve these kind model name, here is an issue opened for this scenario https://github.com/Azure/autorest.csharp/issues/2279

File: sdk/appplatform/ci.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    - feature/*
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/appplatform/
+    - sdk/resourcemanager/
+
+extends:
+  template: ../../eng/pipelines/templates/stages/archetype-sdk-client.yml
```
Comment (benbp) [inline sdk/appplatform/ci.yml:26]:
```suggestion
  template: /eng/pipelines/templates/stages/archetype-sdk-client.yml
```

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/AppResource.cs [hunk 8]
```diff
@@ -0,0 +1,519 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Get an resource upload URL for an App, which may be artifacts or source archive.
+        /// Request Path: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppPlatform/Spring/{serviceName}/apps/{appName}/getResourceUploadUrl
+        /// Operation Id: Apps_GetResourceUploadUrl
+        /// </summary>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<ResourceUploadDefinition>> GetResourceUploadUrlAsync(CancellationToken cancellationToken = default)
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/AppResource.cs:369]: `Uri` is superset of `Url` , it's ok to change from `Url` to \Uri`, but currently the related document still url that makes weird, so I'd like to leave it as is, we can talk with @m-nash for further actions

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/CheckNameAvailabilityContent.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+
+namespace Azure.ResourceManager.AppPlatform.Models
+{
+    /// <summary> Name availability parameters payload. </summary>
+    public partial class CheckNameAvailabilityContent
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/CheckNameAvailabilityContent.cs:13]: Rename to `ServiceNameAvailabilityContent`

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/CheckNameAvailabilityResult.cs [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.AppPlatform.Models
+{
+    /// <summary> Name availability result payload. </summary>
+    public partial class CheckNameAvailabilityResult
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/CheckNameAvailabilityResult.cs:11]: Rename to `ServiceNameAvailabilityResult`

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/ResourceSku.cs (hunk 4 above)
Comment (ArthurMa1978) [reply sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/ResourceSku.cs:14]: Rename to `AvailableAppPlatformSku`

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/SsoProperties.cs (hunk 6 above)
Comment (ArthurMa1978) [reply sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/SsoProperties.cs:39]: Not a `guid`

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/CustomDomainResource.cs [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+
+namespace Azure.ResourceManager.AppPlatform
+{
+    /// <summary>
+    /// A Class representing a CustomDomainResource along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="CustomDomainResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetCustomDomainResource method.
+    /// Otherwise you can get one from its parent resource <see cref="AppResource" /> using the GetCustomDomainResource method.
+    /// </summary>
+    public partial class CustomDomainResource : ArmResource
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/CustomDomainResource.cs:25]: rename to `AppPlatformCustomDomainResource`

File: sdk/appplatform/ci.yml (hunk 7 above)
Comment (ArthurMa1978) [reply sdk/appplatform/ci.yml]: fixed

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/ResourceUploadDefinition.cs [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+
+namespace Azure.ResourceManager.AppPlatform.Models
+{
+    /// <summary> Resource upload definition payload. </summary>
+    public partial class ResourceUploadDefinition
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/ResourceUploadDefinition.cs:13]: According to our guideline, we should avoid using `Definition` as the suffix.

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/DiagnosticParameters.cs [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.AppPlatform.Models
+{
+    /// <summary> Diagnostic parameters of diagnostic operations. </summary>
+    public partial class DiagnosticParameters
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/DiagnosticParameters.cs:11]: Maybe rename this model to `DiagnosticContent` as it is used as the input parameter.

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/AppDeploymentResource.cs [hunk 14]
```diff
@@ -0,0 +1,630 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Get deployment log file URL
+        /// Request Path: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppPlatform/Spring/{serviceName}/apps/{appName}/deployments/{deploymentName}/getLogFileUrl
+        /// Operation Id: Deployments_GetLogFileUrl
+        /// </summary>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<LogFileUriResponse>> GetLogFileUriAsync(CancellationToken cancellationToken = default)
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/AppDeploymentResource.cs:412]: Response is not a good suffix here, can we change it to `LogFileUriResult`?

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/CustomPersistentDiskResource.cs [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+
+namespace Azure.ResourceManager.AppPlatform.Models
+{
+    /// <summary> Custom persistent disk resource payload. </summary>
+    public partial class CustomPersistentDiskResource
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/CustomPersistentDiskResource.cs:13]: This model should not end with the `Resource` suffix.

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/Error.cs [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.AppPlatform.Models
+{
+    /// <summary> The error code compose of code and message. </summary>
+    public partial class Error
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/Error.cs:11]: Single word class name, should rename it.

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/SupportedRuntimePlatform.cs [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private const string JavaValue = "Java";
+        private const string NETCoreValue = ".NET Core";
+
+        /// <summary> Java. </summary>
+        public static SupportedRuntimePlatform Java { get; } = new SupportedRuntimePlatform(JavaValue);
+        /// <summary> .NET Core. </summary>
+        public static SupportedRuntimePlatform NETCore { get; } = new SupportedRuntimePlatform(NETCoreValue);
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/SupportedRuntimePlatform.cs:31]: Do we have exception for `NET` or we also need to rename it to `Net`?

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/SupportedRuntimeValue.cs [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+        private const string NetCore31Value = "NetCore_31";
+
+        /// <summary> Java_8. </summary>
+        public static SupportedRuntimeValue Java8 { get; } = new SupportedRuntimeValue(Java8Value);
+        /// <summary> Java_11. </summary>
+        public static SupportedRuntimeValue Java11 { get; } = new SupportedRuntimeValue(Java11Value);
+        /// <summary> Java_17. </summary>
+        public static SupportedRuntimeValue Java17 { get; } = new SupportedRuntimeValue(Java17Value);
+        /// <summary> NetCore_31. </summary>
+        public static SupportedRuntimeValue NetCore31 { get; } = new SupportedRuntimeValue(NetCore31Value);
```
Comment (Yao725) [inline sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/SupportedRuntimeValue.cs:37]: Same as above.

**Review by Yao725 — APPROVED** [review]
Generally, it looks good to me, still a few minor comments, thanks!

File: sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/Error.cs (hunk 16 above)
Comment (ArthurMa1978) [reply sdk/appplatform/Azure.ResourceManager.AppPlatform/src/Generated/Models/Error.cs]: rename to `AppPlatformErrorInfo`

**Pull request merged**